### ISSUE TITLE
no_unused_imports not working correctly

### ISSUE THREAD

**Issue by ecrmnn** [FriendsOfPHP/PHP-CS-Fixer#4410]
It seems like `php-cs-fixer` do not remove all unused imports.
I have provided a class (Laravel Form Request) as an example.
I'm importing 3 classes (Laravel Models), `User`, `Role`, `App`.
When running `php-cs-fixer` command listed below, it only removes `Role`, but leaves `User` and `App` even though they're not being used either.

---

For configuration or updating questions please read the README and UPGRADE documentation,
or visit: https://gitter.im/PHP-CS-Fixer

When reporting an issue (bug) please provide the following information:

#### The PHP version you are using (`$ php -v`):
=> 7.2.17

#### PHP CS Fixer version you are using (`$ php-cs-fixer -V`):
=> 2.14.0

#### The command you use to run PHP CS Fixer:
=> php-cs-fixer fix app/Admin/Http/Requests/Api/SubscribeRequest.php

#### The configuration file you are using, if any:
```php
<?php

$finder = PhpCsFixer\Finder::create()
    ->exclude('tests/Fixtures')
    ->in(__DIR__);

$config = PhpCsFixer\Config::create()
    ->setRiskyAllowed(true)
    ->setRules([
        '@PSR2' => true,
        'array_indentation' => true,
        'trailing_comma_in_multiline_array' => true,
        'method_chaining_indentation' => true,
        'binary_operator_spaces' => true,
        'strict_comparison' => true,
        'standardize_not_equals' => true,
        'strict_param' => true,
        'concat_space' => ['spacing' => 'one'],
        'array_syntax' => ['syntax' => 'short'],
        'no_unused_imports' => true,
        'no_useless_else' => true,
        'blank_line_before_statement' => true,
        'whitespace_after_comma_in_array' => true,
    ])
    ->setFinder($finder);

return $config;
```

#### If applicable, please provide minimum samples of PHP code (as plain text, not screenshots):
 * before running PHP CS Fixer (no changes):
```php
<?php

namespace App\Admin\Http\Requests\Api;

use App\Models\User;
use App\Models\Role;
use App\Models\App;
use App\Http\Requests\Request;

class SubscribeRequest extends Request
{
    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize()
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules()
    {
        //
    }
}
```

 * with unexpected changes applied when running PHP CS Fixer:
```php
<?php

namespace App\Admin\Http\Requests\Api;

use App\Models\User;
use App\Models\App;
use App\Http\Requests\Request;

class SubscribeRequest extends Request
{
    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize()
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules()
    {
        //
    }
}
```

 * with the changes you expected instead:
```php
<?php

namespace App\Admin\Http\Requests\Api;

use App\Http\Requests\Request;

class SubscribeRequest extends Request
{
    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize()
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules()
    {
        //
    }
}
```

**Comment by julienfalque** [comment]
Are the unused imports removed if you remove the PHPDoc descriptions?

**Comment by ecrmnn** [comment]
No, same behavior without PHPDoc

**Comment by ecrmnn** [comment]
Any tips on how I can get this to work?

**Comment by julienfalque** [comment]
So your snippet has two unused import statements that aren't removed.

`App\Models\App` should be removed but a bug prevents that: `App` appears in the `namespace` declaration, which is considered as a usage by the fixer. This should be fixed by #4416.

`App\Models\User` won't be removed by the fixer because it finds an occurence of the word `user` in a comment. This one is ignored on purpose to prevent breaking applications that might rely on comments. See #2814 for a discussion about how to improve this.

**Comment by ecrmnn** [comment]
Thanks. Any way I can change my config so that it ignores comments?

**Comment by EvKoh** [comment]
> So your snippet has two unused import statements that aren't removed.
> 
> `App\Models\App` should be removed but a bug prevents that: `App` appears in the `namespace` declaration, which is considered as a usage by the fixer. This should be fixed by #4416.
> 
> `App\Models\User` won't be removed by the fixer because it finds an occurence of the word `user` in a comment. This one is ignored on purpose to prevent breaking applications that might rely on comments. See #2814 for a discussion about how to improve this.


Hi Julien, it's Zak, we worked together at HighCo. no_unused_imports doesn't work on traits, I just tested: it works on classes but not traits.

**Comment by julienfalque** [comment]
@EvKoh Hi Zak :) Please submit a new issue with a code snippet that reproduces the problem.
